controllers/authpolicy_authconfig.go

```diff
@@ -235,6 +232,55 @@ func getAffectedPolicies(t *kuadrantgatewayapi.Topology, ap *api.AuthPolicy) []k
 	return affectedPolicies
 }
 
+func policyAsAuthPolicy(p kuadrantgatewayapi.Policy) (*api.AuthPolicy, error) {
+	out := &api.AuthPolicy{}
+	data, err := json.Marshal(p)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, out); err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
```

Comment:
Is there a reason to do a marhsall/unmarshall to convert the type to `*api.AuthPolicy` instead of type casting since it implements `kuadrantgatewayapi.Policy`? Seems like an more expensive operation compared to type casting 🤔

Reply:
Turned out my issue with casting was due to missing the `*` in the type. Now that is fixed I removed this function and put the casting inline. Thanks for calling this out.